src/Workspaces/Core/Portable/SymbolId/SymbolKey.BodyLevelSymbolKey.cs

```diff
@@ -63,6 +63,10 @@ public override SymbolKeyResolution Resolve(Compilation compilation, bool ignore
             private static IEnumerable<ValueTuple<TSymbol, int>> EnumerateSymbols(Compilation compilation, ISymbol containingSymbol, string name, CancellationToken cancellationToken)
             {
                 int ordinal = 0;
+                if (compilation == null)
```

Comment:
We can track that down, but fundamentally `EnumerateSymbols` makes bad assumptions about a compilation being present.  We can:

1. Throw when compilation is null so we make it clear the caller should know that state of the `SymbolKey` object before calling this method
2. Change this method to indicate whether it failed to enumerate symbols or not
3. Have the method return an empty `IEnumerable` as it is being asked enumerate symbols and it has no compilation to get symbols from.